js/gate.js

```diff
@@ -3647,7 +3647,7 @@ module.exports = class gate extends Exchange {
         let filledString = Precise.stringSub (amount, remainingString);
         let cost = this.safeString (order, 'filled_total');
         let rawStatus = undefined;
-        let average = this.safeNumber (order, 'fill_price');
+        let average = this.safeNumber (order, 'avg_deal_price');
```

Comment:
@vladunguru actually for swap orders we should be using `fill_price` 
![image](https://user-images.githubusercontent.com/43336371/220403236-cc475d26-4a0b-41c2-b420-12415b2e361c.png)